scripts/west_commands/runners/jlink.py

```diff
@@ -464,13 +523,36 @@ def run_flash_cmd(self, fname, flash_file, **kwargs):
         if self.supports_loader and self.loader:
             loader_details = "?" + self.loader
 
+        # Determine device type for Commander
+        if self.dev_id:
+            if self.dev_id_type:
+                # Use manually specified device type
+                detected_type = self.dev_id_type
+            else:
+                # Auto-detect device type
+                detected_type = self._detect_dev_id_type(self.dev_id)
+
+            # Build device selection for Commander
+            if detected_type == 'serialno':
+                device_args = ['-USB', f'{self.dev_id}']
+            elif detected_type == 'tty':
+                device_args = ['-USB', f'{self.dev_id}']  # J-Link Commander uses -USB for tty
+            elif detected_type == 'ip':
+                device_args = ['-IP', f'{self.dev_id}']
+            elif detected_type == 'tunnel':
+                device_args = ['-IP', f'{self.dev_id}']  # Treat tunnel as IP
+            else:
+                # Fallback to legacy behavior (USB)
+                device_args = ['-USB', f'{self.dev_id}']
+                self.logger.warning(f'Could not auto-detect device type for "{self.dev_id}",'
+                                  f' using USB as fallback. Use --dev-id-type to specify'
+                                  f' explicitly.')
```

Comment:
I am not sure this message is meaningful for user. In addition, JLink will probably also fail. I suggest a brief message: `f'Could not auto-detect device type for "{self.dev_id}"'`. Or `f'"{self.dev_id}" does not match any known pattern'`.